### PR TITLE
https-addition nginx

### PR DESCRIPTION
**Pull request by onatalushko**
Create https connection for docker nginx

Docker container
https://github.com/skilld-labs/skilld-docker-container/pull/369

### CODE REVIEW

File: 18/Dockerfile [hunk 1]
```diff
@@ -14,6 +14,9 @@ LABEL org.label-schema.build-date=$BUILD_DATE \
 RUN set -ex &&\
   apk add --no-cache --upgrade nginx nginx-mod-http-upload-progress nginx-mod-http-brotli
 
+RUN apk add --no-cache openssl
+RUN openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/ssl-cert-snakeoil.key -out /etc/ssl/certs/ssl-cert-snakeoil.pem -subj "/C=AT/ST=Vienna/L=Vienna/O=Security/OU=Development/CN=example.com"
```
Comment (andypost) [inline 18/Dockerfile:18]: Don't create 2 extra layers in image

File: 18/default.conf [hunk 2]
```diff
@@ -3,6 +3,10 @@ include upstream;
 server {
     server_name SERVER_NAME;
     listen 80;
+    listen 443 ssl http2 default_server;
+    listen [::]:443 ssl http2 default_server;
+    ssl_certificate /etc/ssl/certs/ssl-cert-snakeoil.pem;
+    ssl_certificate_key /etc/ssl/private/ssl-cert-snakeoil.key;
```
Comment (andypost) [inline 18/default.conf:10]: I find it useless in common image

**Review by andypost — CHANGES_REQUESTED** [review]
Please elaborate why you need fake certs!

Moreover not clear why you adding it to common config without limiting ciphers.

Also there's 1.20

**Comment by andypost** [comment]
Thinking a bit more, I find nd all this changes useless.
if your project needs certs just generate it locally and mount into container with required config

**Comment by davidferlay** [comment]
@andypost some third party service work only with https (like payment gateway for commerce projects) so we need to find a way to serve locals in HTTPS

This may or may not be the right implementation but this is definitely useful

**Comment by gaydabura** [comment]
> @andypost some third party service work only with https (like payment gateway for commerce projects) so we need to find a way to serve locals in HTTPS
> 
> This may or may not be the right implementation but this is definitely useful

Traefik + letsencrypt

**Comment by davidferlay** [comment]
We don't use Traefik on locals @gaydabura . Should we ?

**Comment by onatalushko** [comment]
https://github.com/skilld-labs/skilld-docker-container/pull/369

**Comment by gaydabura** [comment]
@davidferlay that may be a solution in your specific case which doesnt require any changes to current project.

**Comment by onatalushko** [comment]
Docker container
https://github.com/skilld-labs/skilld-docker-container/pull/369

**Comment by davidferlay** [comment]
It doesn't sounds like a specific use case to me, I think we should have HTTPS by default in SDC

I propose to update this PR to use Traefik

**Comment by gaydabura** [comment]
@davidferlay traefik should not be used here.
SDC is for 1 project
traefik manage a lot of projects, and every project may use SDC. so better to create documentation how traefik may be used with SDC, but i dont think we must include traefik here, and i am totally sure we should not include ports, certs and others here aswell.

Yes i agree some 3dparty software requires https, but it must not have impact on SDC globally, adding ports and fake certificates will bring us new issues and surely will brake existing projects

**Comment by davidferlay** [comment]
So what's the best course of action to get HTTPS by default in locals ?

**Comment by gaydabura** [comment]
@davidferlay in any case local domain name is a fake name, so means you can not resolve automatic certificate from letsencrypt. But you may generate it locally and use https://doc.traefik.io/traefik/https/tls/#user-defined
So basically it should not be managed in this project but should be as additional suggestion in SDC described in docs. 
In a few words you need
1. docker-compose.yml which will have traefik only in separate folder
2. generated certificates and create traefik.toml 
3. in working folder you run SDC and https://github.com/skilld-labs/skilld-docker-container/blob/master/docker/docker-compose.override.yml.default#L111 as you may see here we have already cert resolver for nginx, but based on dns resolver, i bet its for RA @andypost 
4. So you need to setup traefik to redirect from 80 to 443 on your local domain name and use selfgenerated certificates.

**Comment by andypost** [comment]
@davidferlay that's why I closed it, if something needs this local cert please use volume mount config and certs, please don't bloat image with useless packages

Moreover packaging this extras makes security weird

**Pull request closed without merging**